sonic-thermalctld/scripts/thermalctld

```diff
@@ -0,0 +1,535 @@
+             ('speed', str(speed)),
+             ('speed_tolerance', str(speed_tolerance)),
+             ('speed_target', str(speed_target)),
+             ('led_status', str(try_get(fan.get_status_led))),
+             ('timestamp', datetime.now().strftime('%Y%m%d %H:%M:%S'))
+             ])
+
+        self.table.set(fan_name, fvs)
+
+    def _on_presense_status_changed(self, fan_name, fan_status):
+        """
+        Called when fan presence status changed
+        :param fan_name: Name of the Fan object in case any vendor not implement Fan.get_name
+        :param fan_status: Object representing the FanStatus
+        :return:
+        """
+        if fan_status.presence:
+            logger.log_notice('Fan removed warning cleared: {} was inserted.'.format(fan_name))
+        else:
+            logger.log_warning(
+                'Fan removed warning: {} was removed from the system, potential overheat hazard!'.format(fan_name))
+
+    def _on_under_speed_status_changed(self, fan_name, fan_status):
+        """
+        Called when fan under speed status changed
+        :param fan_name: Name of the Fan object in case any vendor not implement Fan.get_name
+        :param fan_status: Object representing the FanStatus
+        :return:
+        """
+        if not fan_status.under_speed:
+            logger.log_warning('Fan under speed warning cleared: {} speed back to normal.'.format(fan_name))
+        else:
+            logger.log_notice('Fan under speed warning: {} speed over tolerance.'.format(fan_name))
+
+    def _on_over_speed_status_changed(self, fan_name, fan_status):
+        """
+        Called when fan over speed status changed
+        :param fan_name: Name of the Fan object in case any vendor not implement Fan.get_name
+        :param fan_status: Object representing the FanStatus
+        :return:
+        """
+        if not fan_status.over_speed:
+            logger.log_warning('Fan over speed warning cleared: {} speed back to normal.'.format(fan_name))
+        else:
+            logger.log_notice('Fan over speed warning: {} speed over tolerance.'.format(fan_name))
+
+    def _set_fan_led(self, fan, fan_name, fan_status):
+        """
+        Set fan led according to current status
+        :param fan: Object representing a platform Fan
+        :param fan_name: Name of the Fan object in case any vendor not implement Fan.get_name
+        :param fan_status: Object representing the FanStatus
+        :return:
+        """
+        if fan_status.is_ok():
+            fan.set_status_led(FanBase.STATUS_LED_COLOR_GREEN)
+        else:
+            # TODO: wait for Kebo to define the mapping of fan status to led color,
+            # just set it to red so far
+            fan.set_status_led(FanBase.STATUS_LED_COLOR_RED)
+
+
+class TemperStatus(object):
+    def __init__(self):
+        self.over_temper = False
+        self.under_temper = True
+
+    def set_over_temper(self, temper, threshold):
```

Comment:
suggest extracting the common logic in set_under_temper and set_under_temper similar logic to a common function.